ledger/metrics/src/main/scala/com/daml/telemetry/TelemetryContext.scala

```diff
@@ -154,8 +159,10 @@ protected class DefaultTelemetryContext(protected val tracer: Tracer, protected
 
   override def encodeMetadata(): jMap[String, String] = {
     import scala.jdk.CollectionConverters._
-    Tracing.encodeTraceMetadata(Context.current.`with`(span)).asJava
+    Tracing.encodeTraceMetadata(openTelemetryContext).asJava
   }
+
+  override def openTelemetryContext: Context = Context.current.`with`(span)
```

Comment:
what's the reasoning for using Context.current instead of Context.root?

Reply:
There are a few reasons:
- according to the docs 
>It should generally not be required to use the root {@link Context} directly - instead, use {@link Context#current()} to operate on the current {@link Context}. Only use this method if you are absolutely sure you need to disregard the current {@link Context} - this almost always is only a workaround hiding an underlying context propagation issue.
- also as you can see above I'm just refactoring and preserving the current behavior
- if someone would like to use it within `Telemetry.runInOpenTelemetryScope`, it might not work as expected, as the current context would be different